### PR TITLE
Topic/joint composite

### PR DESCRIPTION
**Pull request by gabrielebndn**
Re-implemented calc for joint composite.
I am still checking calc_aba.
Also, although I am sure they are buggy, I didn't touch all sub-methods that will be superseded by LieGroups, except for the neutral configuration method which was necessary for the kinematic tests to run correctly (otherwise the program would segfault when the JointComposite is added to a model).

### CODE REVIEW

File: src/multibody/joint/joint-composite.hxx [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        data.S.matrix().middleCols(idx_v,model.m_nvs[i]) = data.iMlast[succ].inverse().act(jdata.S()); // TODO: avoid computing inverse
+      }
+
+    }
+    
+  };
+
+  inline void jointCompositeCalcZeroOrder(const JointModelComposite & model, JointDataComposite & data, const Eigen::VectorXd & qs)
+  {
+    for (int i=(int)(model.joints.size()-1); i >= 0; --i)
```
Comment (jcarpent) [inline src/multibody/joint/joint-composite.hxx]: Use size_t instead of int

**Comment by gabrielebndn** [comment]
int is necessary to check non-negativity in the loop

File: unittest/joint-composite.cpp [hunk 2]
```diff
@@ -207,5 +212,64 @@ BOOST_AUTO_TEST_CASE(test_copy)
...
+  Model model;
+  JointModelComposite jmodel_composite;
+
+  SE3 config=SE3::Random();
+  //JointModelVariant joint_model;
+  JointIndex parent=0;
+
+  for(int i=0; i<10; i++)
+  {
+    //joint_model = JointModelPX();
```
Comment (jcarpent) [inline unittest/joint-composite.cpp]: Remove these lines if they are not needed.

File: unittest/joint-composite.cpp [hunk 3]
```diff
@@ -207,5 +212,64 @@ BOOST_AUTO_TEST_CASE(test_copy)
...
+
+  Model model_c;
+  model_c.addJoint(0,jmodel_composite,SE3::Identity(),"joint");
+
+  Data data_c(model_c);
+
+  BOOST_CHECK(model.nv == model_c.nv);
+  BOOST_CHECK(model.nq == model_c.nq);
+  
+//  VectorXd ql(ConfigVector_t::Constant(model.nq,-M_PI));
```
Comment (jcarpent) [inline unittest/joint-composite.cpp]: Remove these lines if they are not needed.

**Comment by jcarpent** [comment]
@gabrielebndn Thank you for your contribution.

**Pull request merged**